### ISSUE TITLE
borg stats log output duplicated

### ISSUE THREAD

**Issue by Akito13** [borgbackup/borg#4700]
## Have you checked borgbackup docs, FAQ, and open Github issues?

Yes, thoroughly.

## Is this a BUG / ISSUE report or a QUESTION?

Bug

#### Your borg version (borg -V).
borg 1.1.10
#### Operating system (distribution) and version.
  Operating System: Debian GNU/Linux bullseye/sid
            Kernel: Linux 4.19.0-5-amd64
      Architecture: x86-64

#### Hardware / network configuration, and filesystems used.
* `ext4` client-side
* `zfs` server-side
#### How much data is handled by borg?
does not matter
#### Full borg commandline that lead to the problem (leave away excludes and passwords)
```
sudo borg create --critical -s --comment "Delete me! Just testing." ssh://user@hostname:1234/path/to/server/backup::{hostname}+{user}:{now:%Y-%m-%dT%H:%M} .
```

## Describe the problem you're observing.

I installed borg on a freshly installed Debian, as described above.
Whenever I use the `-s` option, borg is unexpectedly verbose. I can even set the verbosity level to `--critical` and yet, it is super verbose. That said, the verbosity itself is buggy, because it duplicates information without purpose.

**Example:**
```
Number of files: 1
INFO:borg.output.stats:Utilization of max. archive size: 0%
Utilization of max. archive size: 0%
INFO:borg.output.stats:------------------------------------------------------------------------------
------------------------------------------------------------------------------
INFO:borg.output.stats:                       Original size      Compressed size    Deduplicated size
                       Original size      Compressed size    Deduplicated size
INFO:borg.output.stats:This archive:                  844 B                706 B                550 B
This archive:                  844 B                706 B                550 B
INFO:borg.output.stats:All archives:              207.01 GB            168.83 GB             49.68 GB
All archives:              207.01 GB            168.83 GB             49.68 GB
INFO:borg.output.stats:

INFO:borg.output.stats:                       Unique chunks         Total chunks
                       Unique chunks         Total chunks
INFO:borg.output.stats:Chunk index:                  36900              369000
Chunk index:                  36900              369000
INFO:borg.output.stats:------------------------------------------------------------------------------
------------------------------------------------------------------------------

```

#### Can you reproduce the problem? If so, describe how. If not, describe troubleshooting steps you took before opening the issue.
Just back up using borg; use any verbosity level of your choice.

#### Include any warning/errors/backtraces from the system logs
//

**Comment by ThomasWaldmann** [comment]
stuff like requesting stats with `-s` override the loglevel for that stuff, that is intended, because otherwise you wouldn't see it although you requested it.

**Comment by Akito13** [comment]
> 
> 
> stuff like requesting stats with `-s` override the loglevel for that stuff, that is intended, because otherwise you wouldn't see it although you requested it.

Did you look at the code block? I don't think it is intended to output unnecessary redundant information for no purpose. That said, I can prove that this is not normal behaviour, as it works with older version of borgbackup, as intended.

**Comment by ThomasWaldmann** [comment]
OK, so the bug is not that it is verbose, but that stuff is duplicate in output.

**Comment by Akito13** [comment]
> 
> 
> OK, so the bug is not that it is verbose, but that stuff is duplicate in output.

Well, it is also because of the verbosity level, as it shows stuff like `INFO:borg.output.stats:` which should not be shown at all. As it is the case in older borgbackup versions.

**Comment by ThomasWaldmann** [comment]
Works for me:

```
$ borg -V
borg 1.1.10

$ borg init -e none repo

$ borg create -s repo::1 .
------------------------------------------------------------------------------
Archive name: 1
Archive fingerprint: 9d06693d8dfd9a0d761ccf273c17b9b579022d7aac18d4566cd7ce592bf7f6d1
Time (start): Mon, 2019-07-29 21:55:45
Time (end):   Mon, 2019-07-29 21:55:55
Duration: 9.63 seconds
Number of files: 9990
Utilization of max. archive size: 0%
------------------------------------------------------------------------------
                       Original size      Compressed size    Deduplicated size
This archive:              444.26 MB            343.04 MB            330.28 MB
All archives:              444.26 MB            343.04 MB            330.28 MB

                       Unique chunks         Total chunks
Chunk index:                    7142                 7956
------------------------------------------------------------------------------

$ borg create --critical -s repo::2 .
------------------------------------------------------------------------------
Archive name: 2
Archive fingerprint: 7e226074c6fa209b51298f8ac8a8a40fbc9c145c8d8ba944abdbe229dd92e2e3
Time (start): Mon, 2019-07-29 21:56:14
Time (end):   Mon, 2019-07-29 21:56:15
Duration: 1.35 seconds
Number of files: 9990
Utilization of max. archive size: 0%
------------------------------------------------------------------------------
                       Original size      Compressed size    Deduplicated size
This archive:              444.26 MB            343.04 MB            233.17 kB
All archives:              888.52 MB            686.08 MB            330.51 MB

                       Unique chunks         Total chunks
Chunk index:                    7146                15912
------------------------------------------------------------------------------

$ borg create --info -s repo::3 .
Creating archive at "repo::3"
------------------------------------------------------------------------------
Archive name: 3
Archive fingerprint: 92d03af198dcecf3f759b7943e5e0ba43f0d37ecb95f48ba8b39ce79fa7774a6
Time (start): Mon, 2019-07-29 21:56:37
Time (end):   Mon, 2019-07-29 21:56:38
Duration: 1.40 seconds
Number of files: 9990
Utilization of max. archive size: 0%
------------------------------------------------------------------------------
                       Original size      Compressed size    Deduplicated size
This archive:              444.26 MB            343.04 MB                858 B
All archives:                1.33 GB              1.03 GB            330.51 MB

                       Unique chunks         Total chunks
Chunk index:                    7147                23868
------------------------------------------------------------------------------
```

Please give the minimal command sequence that reproduces the issue for you:
- no options not needed to reproduce
- no remote except if needed to reproduce
- no sudo except if needed to reproduce

**Comment by ThomasWaldmann** [comment]
Also I need the output of `set | grep BORG` minus any secrets. For the user and for root.

**Comment by Akito13** [comment]
@ThomasWaldmann 
First of all, thank you for investigating the issue.

I wanted to create a separate repository to test everything as requested, but already when executing
```
borg init --encryption=none .
```
I got this output:
```
ERROR:borg.archiver:A repository already exists at /home/akito/borg-tmp.
A repository already exists at /home/akito/borg-tmp.
```
Apparently, every borg command acts now like this.

**Comment by ThomasWaldmann** [comment]
I've no idea what an empty BORG_TMPDIR would cause, so please either remove that or point it to a valid temp path and retry afterwards.

**Comment by Akito13** [comment]
> 
> 
> I've no idea what an empty BORG_TMPDIR would cause, so please either remove that or point it to a valid temp path and retry afterwards.

It would cause nothing. This setting is set for me since borg 1.1.7 or earlier. I don't remember the exact patch version.

**Comment by ThomasWaldmann** [comment]
I had a look into the source:

yes, it doesn't do anything because borg is not using a `BORG_TMPDIR` environment variable.

I'ld still suggest removing it to not confuse yourself or other people.

**Comment by ThomasWaldmann** [comment]
Please give the minimal command sequence that reproduces the issue described in the top post:

- no options not needed to reproduce
- no remote except if needed to reproduce
- no sudo except if needed to reproduce

**Comment by ThomasWaldmann** [comment]
Any update? See above.

**Comment by Akito13** [comment]
> yes, it doesn't do anything because borg is not using a `BORG_TMPDIR` environment variable.

On my setups it points to the TMPDIR variable. It is an alias, basically.

I tested it several times now under following conditions:

> * no options not needed to reproduce
> 
> * no remote except if needed to reproduce
> 
> * no sudo except if needed to reproduce

Result:
Both options, `--verbose` and `--stats` reproduces this behaviour. Using other options or no options produces normal behaviour.

**Comment by ThomasWaldmann** [comment]
I had already tried with `--info -s` (which is the same thing as `--verbose --stats`) and could not reproduce.

**Comment by Akito13** [comment]
@ThomasWaldmann 
Can it be connected to the Python version? I have
```
Python 3.7.4
```
I doubt this could be the case, as I use almost the same Python version on another device and it works as expected. This one is just one or two patches further.

**Comment by ThomasWaldmann** [comment]
```
$ borg -V
borg 1.1.10

$ borg init -e none newrepo

$ borg create --verbose --stats newrepo::archive .
Creating archive at "newrepo::archive"
------------------------------------------------------------------------------
Archive name: archive
Archive fingerprint: ccb76a6aad8a8b600a3b78c31261db31ba386c0acdb449a83b476e2a290f1136
Time (start): Fri, 2019-08-09 19:16:33
Time (end):   Fri, 2019-08-09 19:16:43
Duration: 10.59 seconds
Number of files: 10072
Utilization of max. archive size: 0%
------------------------------------------------------------------------------
                       Original size      Compressed size    Deduplicated size
This archive:              534.38 MB            432.96 MB            420.20 MB
All archives:              534.38 MB            432.96 MB            420.20 MB

                       Unique chunks         Total chunks
Chunk index:                    7246                 8069
------------------------------------------------------------------------------
```

I can't reproduce. If somebody can and can add more information that can be used to successfully reproduce, feel free to reopen.

**Comment by ThomasWaldmann** [comment]
I also tried on python 3.7.4 (got that via pyenv), can't reproduce.

**Comment by Akito13** [comment]
> 
> 
> I also tried on python 3.7.4 (got that via pyenv), can't reproduce.

Now what can I do to find out the issue? I only experience this problem in borgbackup and I am sure nobody else will take over a problem experienced only in borgbackup.

**Comment by ThomasWaldmann** [comment]
Try it on another machine, on another linux distribution/release, as another user, ... - and if it behaves different, find out what the difference in the environment is.

You can also try using the borg fat binary available from github release, whether that also gives duplicate output.

**Comment by Akito13** [comment]
> find out what the difference in the environment is.

Well, the obvious difference is the Python version, as I had to manually install it, since it is not available over APT, yet, in this Debian version. Everything else basic, no customization really. I just installed `guake` and run everything under KDE. That's it and worked before re-installation, before. The only real difference is the Python version.

> You can also try using the borg fat binary available from github release, whether that also gives duplicate output.
I may try that out, though.
